playbooks/openshift-glusterfs/private/setup_nodes.yml

```diff
@@ -14,5 +14,4 @@
       name: openshift_storage_glusterfs
       tasks_from: kernel_modules.yml
     when:
-    - "inventory_hostname not in groups['glusterfs']"
-    - "inventory_hostname not in groups['glusterfs_registry']"
+    - "oo_glusterfs_to_config is not defined or inventory_hostname not in oo_glusterfs_to_config"
```

Comment:
oo_glusterfs_to_config in groups and inventory_hostname not in groups['oo_glusterfs_to_config']

We only want this to run if 1) we're using gluster (so, oo_glustefs_to_config is a defined group) and 2) we're not a gluster_node, gluster_nodes get setup for this task in another module.

Reply:
I disagree. All nodes should have necessary kernel modules to mount GlusterFS volumes, since a user could be using Gluster volumes from a cluster not within OpenShift. Since the gluster-fuse module is always installed, and the glusterfs mount plugin comes in by default in OpenShift, these kernel modules should always be loaded.

Reply:
@jarrpa if that's the case, we wouldn't be running this playbook at all.  I have already updated the normal install path to include these kernel modules during the node playbook:

https://github.com/openshift/openshift-ansible/blob/master/playbooks/openshift-node/private/configure_nodes.yml#L19

This code is really only needed in the case that gluster support was not added when the cluster was created (not likely), but was added for completeness just in case.

Reply:
So we can just remove this entirely? Or do you want to keep it for the same completeness?

Reply:
I want to keep it for completeness.  This code is only run with running glusterfs playbook directly.  Without the booleans, we might have say a cluster with 100 nodes, 97 of which are gluster hosts, so that would result in 2x tasks for 97 of the nodes, which would be a little slow.

Of course, if you don't care a little about slowness, we can remove the boolean.

Reply:
So if this is only running when running the GlusterFS playbook directly, then that means we are coming from a situation where OpenShift is already installed. We can fairly confidently assume that the glusterfs-fuse client is available and that all nodes have had the client kernel modules configured. Thus, the only change we'd need to make is modify the nodes that will now be GlusterFS server nodes. So we might actually want: 
```
"oo_glusterfs_to_config is defined and inventory_hostname in oo_glusterfs_to_config"
```
Yes?

Reply:
> then that means we are coming from a situation where OpenShift is already installed

Yes, that's what this means.

> We can fairly confidently assume that the glusterfs-fuse client is available and that all nodes have had the client kernel modules configured.

We cannot assume that because it's possible to prevent those things from being configured when installing if the user had no intention of installing gluster (these things would be installed by default, but it's possible for user to prevent).

Now that we've discussed it, I think the change should be
```yaml
"inventory_hostname not in groups['oo_glusterfs_to_config']"
```

Since we're running a gluster-specific play here, oo_glusterfs_to_config will definitely be defined (or the play is useless) and only the non-gluster nodes (eg, nodes that are not in oo_glusterfs_to_config) will need this set of tasks as the oo_glusterfs_to_config hosts/nodes will be setup in another play.